blockchain/blockdao/filedao_new.go

```diff
@@ -0,0 +1,302 @@
+// Copyright (c) 2020 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package blockdao
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/go-pkgs/cache"
+	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+
+	"github.com/iotexproject/iotex-core/action"
+	"github.com/iotexproject/iotex-core/blockchain/block"
+	"github.com/iotexproject/iotex-core/config"
+	"github.com/iotexproject/iotex-core/db"
+	"github.com/iotexproject/iotex-core/db/batch"
+	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
+)
+
+const (
+	_normal             = 0
+	_compressed         = 1
+	blockStoreBatchSize = 16
+)
+
+// namespace for hash, block, and staging storage
+const (
+	hashDataNS    = "hsh"
+	blockDataNS   = "bdn"
+	stagingDataNS = "stg"
+)
+
+var (
+	bottomHeightKey = []byte("bh")
+)
+
+type (
+	// FileDAONew has extra methods for chain db file after file split activation at Ithaca height
+	FileDAONew interface {
```

Comment:
i would say this is not a good name. how would you name the next version of filedao, filedaonewnew?

Reply:
ok, how about fileDAOV2?